### PR TITLE
Fluid api

### LINKED ISSUE #9515: move doc/v2/api/fluid to doc/fluid/api
目前API doc的顺序有点儿乱，请将它们按照如题所示的方法归类。
谢谢！

### LINKED ISSUE #9516: build sphinx tree for fluid API doc
将fluid的API 挪到doc/fluid/api后，请建立sphinx树
谢谢！

### LINKED ISSUE #9517: add compile command for fluid in build_doc.sh
请在这个路径下
paddle/scripts/travis/build_doc.sh 
加入编译fluid的命令
谢谢！

### PR DESCRIPTION
**Pull request by weixing02**
fix #9515
fix #9516
fix #9517

1. Move contents from ```paddle/docs/v2/api/fluid``` to ```paddle/docs/fluid/api``` .

2. Build sphinx tree for ```paddle/docs/fluid/api```.

3. Add commands for compiling fluid (docs/ apis) to ```paddle/scripts/travis/build_doc.sh``` and ```paddle/scripts/docker/build.sh```.

4. Add some dependencies to simplify commands for compiling docs and apis:
- add ```gen_proto_py``` dependencies for ```paddle_v2_docs```, ```paddle_v2_docs_cn```, ```paddle_fluid_docs```, ```paddle_fluid_docs_cn```
- add ```gen_proto_py framework_py_proto copy_paddle_pybind``` dependencies for ```paddle_v2_apis```, ```paddle_fluid_apis```. 

5. Adjust commands structure：
- now we could use ```paddle_docs``` to generate all the documentations.
- now we could use ```paddle_apis``` to generate all the apis.

looks like this:
- ```paddle_docs```:
  - ```paddle_v2_docs```
  - ```paddle_v2_docs_cn```
  - ```paddle_fluid_docs```
  - ```paddle_fluid_docs_cn```
- ```paddle_apis```:
  - ```paddle_v2_apis```
  - ```paddle_fluid_apis```

### CODE REVIEW

File: doc/fluid/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,8 @@ sphinx_add_target(paddle_fluid_docs
                   ${CMAKE_CURRENT_SOURCE_DIR}
                   ${SPHINX_HTML_DIR_EN})
 
+add_dependencies(paddle_fluid_docs gen_proto_py)
```
Comment (luotao1) [inline doc/fluid/CMakeLists.txt]: 这里加的add_dependencies，`make paddle_fluid_docs`前还需要`make gen_proto_py`么？

Comment (weixing02) [reply doc/fluid/CMakeLists.txt]: wx20180330-171316 2x
make paddle_fluid_docs应该和make paddle_v2_docs一样，需要添加make gen_proto_py吧！

Comment (luotao1) [reply doc/fluid/CMakeLists.txt]: 那加在这里没什么意义，可以直接加在`doc/CMakeLists.txt`么？
```
add_custom_target(paddle_docs ALL DEPENDS gen_proto_py paddle_v2_docs xxx)
```
可以对应修改“如何贡献文档”那篇。

File: paddle/scripts/docker/build.sh [hunk 2]
```diff
@@ -127,7 +127,7 @@ EOF
             -DWITH_STYLE_CHECK=OFF
         make -j `nproc` gen_proto_py framework_py_proto
         make -j `nproc` copy_paddle_pybind
-        make -j `nproc` paddle_docs paddle_docs_cn paddle_api_docs
+        make -j `nproc` paddle_docs paddle_apis
```
Comment (luotao1) [inline paddle/scripts/docker/build.sh]: 前面已经加了依赖，这儿128-130行可以只留130行。

File: paddle/scripts/travis/build_doc.sh [hunk 3]
```diff
@@ -9,7 +9,7 @@ cd $TRAVIS_BUILD_DIR/build
 cmake .. -DCMAKE_BUILD_TYPE=Release -DWITH_GPU=OFF -DWITH_MKL=OFF -DWITH_DOC=ON -DWITH_STYLE_CHECK=OFF
 make -j `nproc` gen_proto_py framework_py_proto
 make -j `nproc` copy_paddle_pybind
-make -j `nproc` paddle_docs paddle_docs_cn paddle_api_docs
+make -j `nproc` paddle_docs paddle_apis
```
Comment (luotao1) [inline paddle/scripts/travis/build_doc.sh]: 前面已经加了依赖，这儿10-12行可以只留12行。

File: paddle/scripts/docker/build.sh (hunk 2 above)
Comment (weixing02) [reply paddle/scripts/docker/build.sh]: Done.

File: paddle/scripts/travis/build_doc.sh (hunk 3 above)
Comment (weixing02) [reply paddle/scripts/travis/build_doc.sh]: Done.

**Pull request merged**